web3/main.py

```diff
@@ -173,9 +173,15 @@ def keccak(primitive=None, text=None, hexstr=None):
         )
 
     @combomethod
+    @deprecated_for("solidityKeccak")
     def soliditySha3(cls, abi_types, values):
+        # just call it as a class method
+        return Web3.solidityKeccak(abi_types, values)
```

Comment:
So I believe forcing this to use the class breaks ENS when using the deprecated method: `w3.soliditySha3(['address'], ['ethereumfoundation.eth'])` (Note new lines 194-198)

Apparently there's no test for it, which isn't hugely surprising, since a full test involves having ENS deployed with a name registered.

Switching this to `cls.solidityKeccak`... should resolve it AFAICT.

Reply:
Pushed commit b05e942 with recommendation.